pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -374,6 +374,42 @@ async def _build_content_and_config(
         )
         return contents, config
 
+    @staticmethod
+    def _map_finish_reason_to_otel(raw: str | None) -> str | None:
+        """Map provider-specific finish reasons to OpenTelemetry gen_ai.response.finish_reasons values.
+
+        Only returns a value if it matches a known OTEL value; otherwise returns None.
+        """
+        if raw is None:
+            return None
+        upper = raw.upper()
```

Comment:
Do we need this? According to the types, the values will always be uppercase already